### PR TITLE
Helm chart for Kusk Gateway

### PR DESCRIPTION
**Pull request by Tarick**
## Pull request description 

Kusk Gateway chart

## Checklist (choose whats happened)

- [ ] breaking change! (describe)
- [x] tested locally
- [ ] tested on cluster
- [ ] added new dependencies
- [ ] updated the docs
- [ ] added a test

## Breaking changes

-

## Changes

1. Adds kusk-gateway and kusk-gateway-envoyfleet charts. Kusk gateway chart has subchart with CRDs due to the problem of installing and uninstalling CRDs with helm. kusk-gateway-envoyfleet chart is needed since the CR resource can't be installed with Helm until CRDs are installed and Helm can validate the resource. Specifying ```--disable-openapi-validation``` doesn't help here.
Common variables are passed from main chart to subchart with templates in _helpers - this way we can still use values.yaml variables in template generation and not use "global" variables block.

2. Changes release workflow to separate flows for Testkube and Kusk Gateway. This potentially can cause simultaneous runs, maybe we can trigger Testkube tests workflows separately with [workflow_run](https://docs.github.com/en/actions/learn-github-actions/events-that-trigger-workflows#workflow_run)? @fog1985 

## Fixes

-

### CODE REVIEW

File: charts/kusk-gateway-envoyfleet/templates/_helpers.tpl [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+{{- end }}
+
+{{/*
+Selector labels
+*/}}
+{{- define "kusk-gateway-envoyfleet.selectorLabels" -}}
+app.kubernetes.io/name: {{ include "kusk-gateway-envoyfleet.name" . }}
+app.kubernetes.io/instance: {{ .Release.Name }}
+
+{{- end }}
```
Comment (kylehodgetts) [inline charts/kusk-gateway-envoyfleet/templates/_helpers.tpl:53]: newline

File: charts/kusk-gateway-envoyfleet/templates/envoyFleet.yaml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+---
+apiVersion: gateway.kusk.io/v1alpha1
+kind: EnvoyFleet
+metadata:
+  name: {{ .Values.name }}
+  labels:
+    {{- include "kusk-gateway-envoyfleet.labels" . | nindent 4 }}
+spec:
+  size: {{ .Values.size }}
```
Comment (kylehodgetts) [inline charts/kusk-gateway-envoyfleet/templates/envoyFleet.yaml:9]: newline

File: charts/kusk-gateway-envoyfleet/values.yaml [hunk 3]
```diff
@@ -0,0 +1,10 @@
+# Default values
+# This is a YAML-formatted file.
+# Declare variables to be passed into your templates.
+
+nameOverride: ""
+fullnameOverride: ""
+
+# Fleet name, i.e. Fleet ID
+name: default
+size: 1
```
Comment (kylehodgetts) [inline charts/kusk-gateway-envoyfleet/values.yaml:10]: newline

File: charts/kusk-gateway/charts/kusk-gateway-crds/values.yaml [hunk 4]
```diff
@@ -0,0 +1,6 @@
+# Default values for kusk-gateway-crds.
+# This is a YAML-formatted file.
+# Declare variables to be passed into your templates.
+
+nameOverride: ""
+fullnameOverride: ""
```
Comment (kylehodgetts) [inline charts/kusk-gateway/charts/kusk-gateway-crds/values.yaml:6]: newline

File: charts/kusk-gateway/templates/certmanagement.yaml [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+---
+apiVersion: cert-manager.io/v1
+kind: Issuer
+metadata:
+  name: kusk-gateway-selfsigned-issuer
+  namespace: {{ .Release.Namespace }}
+spec:
+  selfSigned: {}
+
+{{- end -}}
```
Comment (kylehodgetts) [inline charts/kusk-gateway/templates/certmanagement.yaml:25]: newline

File: charts/kusk-gateway/templates/deployment.yaml [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+      {{- with .Values.tolerations }}
+      tolerations:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      terminationGracePeriodSeconds: {{ .Values.terminationGracePeriodSeconds }}
+      volumes:
+      - name: cert
+        secret:
+          defaultMode: 420
+          secretName: "{{ include "kusk-gateway.webhooksCertificate" . }}"
```
Comment (kylehodgetts) [inline charts/kusk-gateway/templates/deployment.yaml:102]: newline

File: charts/kusk-gateway/templates/poddisruptionbudget.yaml [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+metadata:
+  name: kusk-gateway
+  labels:
+    {{- include "kusk-gateway.labels" . | nindent 4 }}
+spec:
+{{ toYaml .Values.podDisruptionBudget | indent 2 -}}
+  selector:
+    matchLabels:
+      {{- include "kusk-gateway.selectorLabels" . | nindent 6 }}
+{{- end }}
```
Comment (kylehodgetts) [inline charts/kusk-gateway/templates/poddisruptionbudget.yaml:14]: newline

File: charts/kusk-gateway/templates/rolebinding.yaml [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: Role
+  name: kusk-gateway-leader-election-role
+subjects:
+  - kind: ServiceAccount
+    name: {{ template "kusk-gateway.serviceAccountName" . }}
+    namespace: {{ .Release.Namespace | quote }}
+
+{{- end -}}
```
Comment (kylehodgetts) [inline charts/kusk-gateway/templates/rolebinding.yaml:50]: newline

File: charts/kusk-gateway/templates/roles.yaml [hunk 9]
```diff
@@ -0,0 +1,339 @@
...
+    verbs:
+      - create
+      - delete
+      - get
+      - list
+      - patch
+      - update
+      - watch
+
+{{- end -}}
```
Comment (kylehodgetts) [inline charts/kusk-gateway/templates/roles.yaml:339]: newline

File: charts/kusk-gateway/templates/service.yaml [hunk 10]
```diff
@@ -0,0 +1,37 @@
...
+  type: {{ .Values.webhooks.service.type }}
+  ports:
+    - port: {{ .Values.webhooks.service.port }}
+      targetPort: webhook-server
+      protocol: TCP
+      name: webhook-server
+  selector:
+    {{- include "kusk-gateway.selectorLabels" . | nindent 4 }}
+
+{{- end -}}
```
Comment (kylehodgetts) [inline charts/kusk-gateway/templates/service.yaml:37]: newline

File: charts/kusk-gateway/templates/serviceaccount.yaml [hunk 11]
```diff
@@ -0,0 +1,12 @@
...
+kind: ServiceAccount
+metadata:
+  name: {{ include "kusk-gateway.serviceAccountName" . }}
+  labels:
+    {{- include "kusk-gateway.labels" . | nindent 4 }}
+  {{- with .Values.serviceAccount.annotations }}
+  annotations:
+    {{- toYaml . | nindent 4 }}
+  {{- end }}
+{{- end }}
```
Comment (kylehodgetts) [inline charts/kusk-gateway/templates/serviceaccount.yaml:12]: newline

File: charts/kusk-gateway/templates/webhooks.yaml [hunk 12]
```diff
@@ -0,0 +1,106 @@
...
+    - v1alpha1
+    operations:
+    - CREATE
+    - UPDATE
+    resources:
+    - staticroutes
+  sideEffects: None
+
+
+{{- end -}}
```
Comment (kylehodgetts) [inline charts/kusk-gateway/templates/webhooks.yaml:106]: newline

File: charts/kusk-gateway/values.yaml [hunk 13]
```diff
@@ -0,0 +1,113 @@
...
+    #   cpu: 100m
+    #   memory: 128Mi
+
+# KubeRBACProxy container settings
+kubeRBACProxy:
+  resources: {}
+
+terminationGracePeriodSeconds: 10
+
+podDisruptionBudget: {}
```
Comment (kylehodgetts) [inline charts/kusk-gateway/values.yaml:113]: newline

**Review by kylehodgetts — CHANGES_REQUESTED** [review]
The content looks great! Just Github complaining about new lines in the YAML files

**Review by kylehodgetts — APPROVED** [review]
Preliminarily approved. Will test on Monday 👍 
Good work!

**Review by dobegor — APPROVED** [review]

**Review by fog1985 — APPROVED** [review]
LGTM

**Pull request merged**